src/main/scala/at/logic/skeptik/parser/ParserTPTP.scala

```diff
@@ -223,20 +223,33 @@ extends TokenParsers with PackratParsers {
 
   def term: Parser[E] = (
     function_term
-      ||| variable         ^^ {x => {recordVar(x);Variable(x)}}
+      ||| variable         ^^ {stringToVariable(_)}
       ||| conditional_term
       ||| let_term
     )
 
+  private def stringToVariable(x: String) : E =
+    if(typedExpressions contains x) {
+      recordVar(x,typedExpressions(x))
+      TypedVariable(x,typedExpressions(x))
+    } else {
+      recordVar(x)
+      Variable(x)
+    }
+
   def function_term: Parser[E] = (
```

Comment:
On the other hand, I have just seen that you use your convention everywhere in your parsers... It would be a lot of work to change that. I guess it is ok to leave it like that. But please stick to the Scala conventions outside the parsers.